src/jailer/src/env.rs

```diff
@@ -184,6 +189,41 @@ impl Env {
         self.uid
     }
 
+    fn exec_into_new_pid_ns(&self, chroot_exec_file: PathBuf) -> Result<()> {
+        // Unshare into a new PID namespace.
+        // The current process will not be moved into the newly created namespace, but its first
+        // child will assume the role of init(1) in the new namespace.
+        // The call is safe because we're invoking a C library function with valid parameters.
+        SyscallReturnCode(unsafe { libc::unshare(libc::CLONE_NEWPID) })
+            .into_empty_result()
+            .map_err(Error::UnshareNewPID)?;
+
+        // Duplicate the current process. The child process will belong to the previously created
+        // PID namespace.
+        let pid = unsafe { libc::fork() };
```

Comment:
`libc::fork()` might be a big hammer here, can we do it by calling the underlying `clone()` syscall with the right set of args ourselves?